File: 0917-reverse-only-letters/0917-reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        left, right = 0, len(s) - 1
        reverse_s = [c for c in s]
        
        while left < right:
            if (s[left].isalpha() and s[right].isalpha()):
                reverse_s[left], reverse_s[right] = reverse_s[right], reverse_s[left]
                left += 1
                right -= 1
            elif not s[left].isalpha():
                left += 1
            elif not s[right].isalpha():
                right -= 1
            
        return "".join(reverse_s)

File: 0917-reverse-only-letters/0917-reverse-only-letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int left = 0, right = s.size() - 1;
        
        while (left < right){
            if (isalpha(s[left]) && isalpha(s[right])){
                swap(s[left], s[right]);
                ++left;
                --right;
            }
            else if (!isalpha(s[left])) ++left;
            else if (!isalpha(s[right])) --right;
        }
        
        return s;
    }
};